src/main/java/org/osiam/resources/exceptions/OsiamExceptionHandler.java

```diff
@@ -23,68 +23,99 @@
 
-    /**
-     * Contains all known ErrorMessageTransformer to validate and manipulate error messages
-     */
-    private static final ErrorMessageTransformer[] KNOWN_ERROR_MSG_TRANSFORMER = {
-            new TypeErrorMessageTransformer(),
-            new JsonPropertyMessageTransformer(),
-            new JsonMappingMessageTransformer() };
-
-    @ExceptionHandler(value = { Exception.class })
-    protected ResponseEntity<Object> handleConflict(Exception ex, WebRequest request) {
-        LOGGER.warn("An exception occurred", ex);
-        HttpStatus status = setStatus(ex);
-        JsonErrorResult error = new JsonErrorResult(status.name(), constructMessage(ex.getMessage()));
-        return handleExceptionInternal(ex, error, new HttpHeaders(), status, request);
+    @ExceptionHandler(Exception.class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    @ResponseBody
+    public JsonErrorResult defaultExceptionHandler(Exception ex) {
+        LOGGER.warn("An unexpected exception occurred", ex);
+        return new JsonErrorResult(HttpStatus.CONFLICT.name(), "An unexpected error occurred");
     }
 
-    /**
-     * This method may transform error messages to a more human readable message if the message contains some known
-     * buzzwords to trigger the effect.
-     *
-     * @param message
-     *            the message to check and transform
-     * @return the original message or a transformed message
-     */
-    private String constructMessage(String message) {
-        String result = message;
-        for (ErrorMessageTransformer et : KNOWN_ERROR_MSG_TRANSFORMER) {
-            result = et.transform(result);
-        }
-        return result;
+    @ExceptionHandler({IllegalArgumentException.class, InvalidConstraintException.class})
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ResponseBody
+    public JsonErrorResult handleInvalidConstraintException(Exception e) {
+        return produceErrorResponse(e.getMessage(), HttpStatus.BAD_REQUEST);
     }
 
-    private HttpStatus setStatus(Exception ex) {
-        if (ex instanceof ResourceNotFoundException) {
-            return HttpStatus.NOT_FOUND;
-        }
-        if (ex instanceof SchemaUnknownException) {
-            return HttpStatus.I_AM_A_TEAPOT;
-        }
-        if (ex instanceof UnsupportedOperationException) {
-            return HttpStatus.NOT_IMPLEMENTED;
-        }
+    @ExceptionHandler(ResourceExistsException.class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    @ResponseBody
+    public JsonErrorResult handleResourceExists(ResourceExistsException e) {
+        return produceErrorResponse(e.getMessage(), HttpStatus.CONFLICT);
+    }
+
+    @ExceptionHandler(OsiamBackendFailureException.class)
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    @ResponseBody
+    public JsonErrorResult handleBackendFailure(OsiamBackendFailureException e) {
+        return produceErrorResponse("An internal error occurred", HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
That's right. I removed the message from the exception and log at the origin now.